lib/mongo/error/operation_failure.rb

```diff
@@ -18,6 +18,35 @@ class Error
     # Raised when an operation fails for some reason.
     #
     # @since 2.0.0
-    class OperationFailure < Error; end
+    class OperationFailure < Error
+
+      # These are magic error messages that could indicate a cluster
+      # reconfiguration behind a mongos. We cannot check error codes as they
+      # change between versions, for example 15988 which has 2 completely
+      # different meanings between 2.4 and 3.0.
+      #
+      # @since 2.1.1
+      RETRY_MESSAGES = [
```

Comment:
`"could not get last error"` and `"connection attempt failed"` and `""` are others I think you should add here.